### ISSUE TITLE
Space before context bound colon only when there is more than one

### ISSUE THREAD

**Issue by durban** [scalameta/scalafmt#916]
Feature request: space before context bound colon only when there is more than one

- **Version**: 0.6.8
- **Integration**: sbt
- **Configuration**: default

## Problem

Scalafmt formats code like this:
```scala
final class Test[A: Monoid] {}
final class More[A: Monoid: Eq] {}
final class Sup[A <: Foo: Eq] {}
final class Sub[A >: Foo: Eq] {}
final class Lot[A >: Foo: Eq: Monoid] {}
```

With `spaces.beforeContextBoundColon = true` it formats code like this:
```scala
final class Test[A : Monoid] {}
final class More[A : Monoid : Eq] {}
final class Sup[A <: Foo : Eq] {}
final class Sub[A >: Foo : Eq] {}
final class Lot[A >: Foo : Eq : Monoid] {}
```

## Expectation

I would like the formatted output to look like this:
```scala
final class Test[A: Monoid] {}
final class More[A : Monoid : Eq] {}
final class Sup[A <: Foo : Eq] {}
final class Sub[A >: Foo : Eq] {}
final class Lot[A >: Foo : Eq : Monoid] {}
```

That is, I would *not* like a space before a context bound colon if and only if the type parameter has only one bound. If it has more than one bound (any kind, not just context), I would like a space before the context bound colon.

## Notes

This preference was already mentioned [here](https://github.com/scalameta/scalafmt/issues/180#issuecomment-211392615) in #180, but it seems it wasn't implemented.

**Comment by olafurpg** [comment]
Thank you for reporting! I'm open to add an option for this, considering the interest from Gitter.

If someone wants to take a stab at this, a good place to start would be here
https://github.com/scalameta/scalafmt/blob/d5c3b0a14c8abb0f6afb9d95b50a67856afd9dba/scalafmt-core/shared/src/main/scala/org/scalafmt/internal/Router.scala#L797

What would be a good name for this option?

**Comment by durban** [comment]
Is it possible to make `spaces.beforeContextBoundColon` take an enum instead of a boolean? If yes, it could be something like `always` (the current `true`), `never` (the current `false`) and `ifMultipleBounds`, for example.

**Comment by olafurpg** [comment]
This was fixed in #956
